File: MiniDOMWebLibs/MiniDOM/Session/CSiteSessionsCursor.cs
using System;

namespace minidom
{
    public partial class WebSite
    {
        public class CSiteSessionsCursor : Databases.DBObjectCursorBase<CSiteSession>
        {
            private Databases.CCursorFieldObj<string> m_SessionID = new Databases.CCursorFieldObj<string>("SessionID");
            private Databases.CCursorField<DateTime> m_StartTime = new Databases.CCursorField<DateTime>("StartTime");
            private Databases.CCursorField<DateTime> m_EndTime = new Databases.CCursorField<DateTime>("EndTime");
            private Databases.CCursorFieldObj<string> m_RemoteIP = new Databases.CCursorFieldObj<string>("RemoteIP");
            private Databases.CCursorField<int> m_RemotePort = new Databases.CCursorField<int>("RemotePort");
            private Databases.CCursorFieldObj<string> m_UserAgent = new Databases.CCursorFieldObj<string>("UserAgent");
            private Databases.CCursorFieldObj<string> m_Cookie = new Databases.CCursorFieldObj<string>("Cookie");
            private Databases.CCursorFieldObj<string> m_InitialReferrer = new Databases.CCursorFieldObj<string>("InitialReferrer");

            public CSiteSessionsCursor()
            {
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Databases.LOGConn;
            }

            protected override Sistema.CModule GetModule()
            {
                return Sessions.Module;
            }

            public override string GetTableName()
            {
                return "tbl_SiteSessions";
            }

            public Databases.CCursorFieldObj<string> SessionID
            {
                get
                {
                    return m_SessionID;
                }
            }

            public Databases.CCursorField<DateTime> StartTime
            {
                get
                {
                    return m_StartTime;
                }
            }

            public Databases.CCursorField<DateTime> EndTime
            {
                get
                {
                    return m_EndTime;
                }
            }

            public Databases.CCursorFieldObj<string> RemoteIP
            {
                get
                {
                    return m_RemoteIP;
                }
            }

            public Databases.CCursorField<int> RemotePort
            {
                get
                {
                    return m_RemotePort;
                }
            }

            public Databases.CCursorFieldObj<string> UserAgent
            {
                get
                {
                    return m_UserAgent;
                }
            }

            public Databases.CCursorFieldObj<string> Cookie
            {
                get
                {
                    return m_Cookie;
                }
            }

            public Databases.CCursorFieldObj<string> InitialReferrer
            {
                get
                {
                    return m_InitialReferrer;
                }
            }
        }
    }
}

File: MiniDOMWebLibs1/MiniDOM/Session/CSiteSessionsCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema

Partial Class WebSite

    Public Class CSiteSessionsCursor
        Inherits DBObjectCursorBase(Of CSiteSession)

        Private m_SessionID As New CCursorFieldObj(Of String)("SessionID")
        Private m_StartTime As New CCursorField(Of Date)("StartTime")
        Private m_EndTime As New CCursorField(Of Date)("EndTime")
        Private m_RemoteIP As New CCursorFieldObj(Of String)("RemoteIP")
        Private m_RemotePort As New CCursorField(Of Integer)("RemotePort")
        Private m_UserAgent As New CCursorFieldObj(Of String)("UserAgent")
        Private m_Cookie As New CCursorFieldObj(Of String)("Cookie")
        Private m_InitialReferrer As New CCursorFieldObj(Of String)("InitialReferrer")

        Public Sub New()
        End Sub

        Protected Overrides Function GetConnection() As CDBConnection
            Return Databases.LOGConn
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Sessions.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_SiteSessions"
        End Function

        Public ReadOnly Property SessionID As CCursorFieldObj(Of String)
            Get
                Return Me.m_SessionID
            End Get
        End Property

        Public ReadOnly Property StartTime As CCursorField(Of Date)
            Get
                Return Me.m_StartTime
            End Get
        End Property

        Public ReadOnly Property EndTime As CCursorField(Of Date)
            Get
                Return Me.m_EndTime
            End Get
        End Property

        Public ReadOnly Property RemoteIP As CCursorFieldObj(Of String)
            Get
                Return Me.m_RemoteIP
            End Get
        End Property

        Public ReadOnly Property RemotePort As CCursorField(Of Integer)
            Get
                Return Me.m_RemotePort
            End Get
        End Property

        Public ReadOnly Property UserAgent As CCursorFieldObj(Of String)
            Get
                Return Me.m_UserAgent
            End Get
        End Property

        Public ReadOnly Property Cookie As CCursorFieldObj(Of String)
            Get
                Return Me.m_Cookie
            End Get
        End Property

        Public ReadOnly Property InitialReferrer As CCursorFieldObj(Of String)
            Get
                Return Me.m_InitialReferrer
            End Get
        End Property

    End Class


End Class
